keras_nlp/models/t5/t5_backbone.py

```diff
@@ -0,0 +1,846 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""T5 backbone model."""
+
+import copy
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.compiler.tf2xla.python.xla import dynamic_slice
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.backbone import Backbone
+from keras_nlp.models.t5.t5_presets import backbone_presets
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras_nlp_export("keras_nlp.models.T5Backbone")
+class T5Backbone(Backbone):
+    def __init__(
+        self,
+        vocabulary_size,
+        output_dim,
```

Comment:
> Lastly "num_blocks" we call "num_layers". Again it's a place there is no perfect consistency, but here is BERT referring to a 12 block model as a 12 layer model 

I disagree with this because the concept of Layer is pretty well established in Keras and is a lot more atomic than this.

Reply:
I applied the other name substitutions and remove `key_value_dim`.